src/components/textinput.js

```diff
@@ -204,10 +206,16 @@
       handleValidation(validity);
     };
 
+    const focusHandler = () =>
+      setTimeout(() => {
+        inputRef.current.focus();
+      }, 50);
```

Comment:
Perhaps its better to use `0` instead `50`.
See https://developer.mozilla.org/en-US/docs/Web/API/setTimeout
Or do you have a specific reason to use `50`?

Reply:
Sweet, didn't know about that. A timeout of `0` seemed to work consistently in my test cases as well. Added a new commit to this PR.